### ISSUE TITLE
Md-toast should be accessible

### ISSUE THREAD

**Issue by marcysutton** [angular/material#688]
One of the more complex interactions, `<md-toast>`, needs to be made accessible. Because this is an interim element like `<md-dialog>`, it shares common characteristics and code, with the exception that md-toast goes away automatically after a timeout.

What needs to happen with toast: 
1. Focus should be sent to the toast or toast child action for keyboard access
2. Toasts should be read aloud in a screen reader when they appear, either by using `aria-live` on a parent element or by sending focus to it
3. Focus should be sent back to triggering element after toast disappears/closes.

Sending focus to the element seems to make the most sense, since adding `aria-live` to the toast's parent element puts responsibility outside of the component itself. Adding `tabindex="-1"` will make the toast focusable by script.
